src/frontend/src/components/dock.js

```diff
@@ -122,8 +123,22 @@ class Dock extends PureComponent<Props> {
       }
     }
 
-    profile.favorites = favorites;
-    handleAuthChange(profile);
+    // Save updated profile
+    const updatedProfile = { ...profile, favorites };
+    if (profile.voucherNumber === ANONYMOUS_USERNAME) {
+      // Despite the name, this saves the profile to local storage in anonymous mode.
+      this.props.handleAuthChange(updatedProfile);
+    } else {
+      // Save to API
+      this.props
+        .saveProfile(updatedProfile, this.props.authToken)
+        .then((newProfile) => {
+          this.props.setProfile(newProfile);
+        })
+        .catch((err) => {
```

Comment:
This isn't very friendly if it fails but I wasn't sure of the best way to visually communicate a failure here, so I opted to simply have it not work, and in most cases we'll get a stack trace in the Django logs to let us know something is wrong.

Reply:
Since there's Django logs I think that's the right decision. The star remaining unfilled after selection feels like an immediate/straightforward enough indicator to either try again or signal it's not working as it's supposed to, so I think it covers our bases enough in that respect.